tracer/test/Datadog.Trace.Tests/Configuration/GlobalSettingsTests.cs

```diff
@@ -18,39 +22,42 @@ public class GlobalSettingsTests : SettingsTestsBase
         public void DebugEnabled(string value, bool expected)
         {
             var source = CreateConfigurationSource((ConfigurationKeys.DebugEnabled, value));
-            var settings = new GlobalSettings(source, NullConfigurationTelemetry.Instance);
+            var settings = new GlobalSettings(source, NullConfigurationTelemetry.Instance, new OverrideErrorLog());
 
             settings.DebugEnabled.Should().Be(expected);
         }
 
         [Theory]
-        [InlineData("true", "info", true)]
-        [InlineData("true", "debug", true)]
-        [InlineData("false", "info", false)]
-        [InlineData("false", "debug", false)]
-        [InlineData("A", "info", false)]
-        [InlineData("A", "debug", false)]
-        [InlineData(null, "info", false)]
-        [InlineData(null, "debug", true)]
-        [InlineData("", "info", false)]
-        [InlineData("", "debug", false)]
-        public void OtelLogLevelDebugSetsDebugEnabled(string value, string otelValue, bool expected)
+        [InlineData("true", "info", true, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        [InlineData("true", "debug", true, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        [InlineData("false", "info", false, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        [InlineData("false", "debug", false, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        [InlineData("A", "info", false, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        [InlineData("A", "debug", false, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        [InlineData(null, "info", false, (int)Count.OpenTelemetryConfigInvalid)]
+        [InlineData(null, "debug", true, null)]
+        [InlineData("", "info", false, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        [InlineData("", "debug", false, (int)Count.OpenTelemetryConfigHiddenByDatadogConfig)]
+        public void OtelLogLevelDebugSetsDebugEnabled(string value, string otelValue, bool expected, int? metric)
         {
             const string otelKey = ConfigurationKeys.OpenTelemetry.LogLevel;
             var source = CreateConfigurationSource(
                 (ConfigurationKeys.DebugEnabled, value),
                 (otelKey, otelValue));
-            var settings = new GlobalSettings(source, NullConfigurationTelemetry.Instance);
+            var errorLog = new OverrideErrorLog();
+            var settings = new GlobalSettings(source, NullConfigurationTelemetry.Instance, errorLog);
 
             settings.DebugEnabled.Should().Be(expected);
+            // Is DD_LOG_LEVEL correct here? Shouldn't it be ConfigurationKeys.DebugEnabled?
```

Comment:
Is this question a blocker to merging?

Reply:
> Is this question a blocker to merging?

It's one of many blockers, it's a question for @zacharycmontoya - I expect it's right (as that's what it was before), but just want to confirm

Reply:
Thanks for double-checking, this is a strange one in the OTEL mapping. The rule goes like this: `OTEL_LOG_LEVEL` sets `DD_LOG_LEVEL`, and if the value was `debug` then `DD_TRACE_DEBUG` is also enabled.

I actually don't know if we should be emitting telemetry for an invalid scenario when only setting `DD_TRACE_DEBUG`. I'll follow up on that separately

Reply:
FWIW the rest of the PR looks good for me, so if it's not too inconvenient to keep this PR open, we can fix this in the same PR once we get clarity on what's expected here

Reply:
Happy to wait for the answer, we also need to wait for these outstanding tasks (specifically the second one)

- [ ] The metrics aren't added to the intake yet, so need updating there (and copying here)
- [ ] The tag names aren't "valid" for standard telemetry metrics, they should be lowercase, and shouldn't use .. I would suggest changing these to be lowercase and replacing . with _, but either way this should happen after they've been OK'd in the intake.

Reply:
OK, so just to be clear, the following line should be:

```csharp
errorLog.ShouldHaveExpectedOtelMetric(metric, ConfigurationKeys.OpenTelemetry.LogLevel, ConfigurationKeys.DebugEnabled);
```

instead of `DD_LOG_LEVEL`, right?

Reply:
Yes exactly!

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5717/commits/940a09e3d35a4225e75da18402979d33700b5f61.

We still have the issue with the telemetry metric tags being non-standard though; I think we should normalise them by replacing `.` with `_` in the "name" part, and lowercasing the whole thing. What do you think @zacharycmontoya? Are you ok with that? That's the standard all other telemetry-metric tags use currently...